macOS Sonoma Build Bugs

### Describe the bug

I'm doing some test builds on a macOS Sonoma VM and have found some issues with the process. Documenting them here, potentially fixing some.  I compiled Mac Ports from source as there is no Sonoma package available yet. This is a clean install of Sonoma Beta 8 for the sole purpose of testing darktable building. Figured it's a good time to get out in front of this and test it out before Sonoma proper is released.

### Steps to reproduce

1. Follow instructions in packaging/macosx/build.txt file after a checkout of release-4.4.2 (also testing master)
2. Build fails on macOS Sonoma
3. First error was libomp wasn't detected:
`CMake Error at /opt/local/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES)
  (Required is at least version "4.5")
Call Stack (most recent call first):
  /opt/local/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /opt/local/share/cmake-3.24/Modules/FindOpenMP.cmake:545 (find_package_handle_standard_args)
  CMakeLists.txt:90 (find_package)`
 
This was fixed with `port install libomp` so I think that should probably be added to the package list in the documentation maybe?

Second error I cannot get past is:

```
-- Missing intltool-merge
-- Missing desktop-file-validate, problems in org.darktable.darktable.desktop might go unnoticed
-- Could NOT find LLVM (missing: LLVM_DIR)
CMake Warning at CMakeLists.txt:352 (message):
  Could not find LLVM 7 or above


CMake Warning at CMakeLists.txt:353 (message):
  Test-compilation of OpenCL programs can not be done.
```

I've got Xcode beta 15 installed and the command line tools beta, clang reports the following:

```
% clang --version
Apple clang version 15.0.0 (clang-1500.0.40.1)
Target: arm64-apple-darwin23.0.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```



### Expected behavior

produce dmg file and .app package

### Where did you install darktable from?

self compiled

### darktable version

4.4.2

### What OS are you using?

Mac

### What is the version of your OS?

macOS 14 Sonoma

### Describe your system?

VM on M2 Max

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

M2 Max

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

_No response_

--- a/packaging/macosx/3_make_hb_darktable_package.sh
+++ b/packaging/macosx/3_make_hb_darktable_package.sh
@@ -85,7 +85,7 @@ function reset_exec_path {
         oToolLoader=$(otool -l "$1" 2>/dev/null | grep '@loader_path' | cut -d\( -f1 | sed 's/^[[:blank:]]*path[[:blank:]]*//;s/[[:blank:]]*$//' )
         if [[ "$oToolLoader" == "@loader_path/../lib/darktable" ]]; then
             echo "Resetting loader path for libdarktable.dylib of <$1>"
-            install_name_tool -rpath @loader_path/../lib/darktable @loader_path/../Resources/lib/darktable "$1"
+            install_name_tool -rpath @loader_path/../lib/darktable @loader_path/../Resources/lib/darktable "$1" || true
         fi
     fi
 
@@ -110,14 +110,14 @@ function reset_exec_path {
             echo "Resetting executable path for dependency <$hbDependency> of <$1>"
 
             # Set correct executable path
-            install_name_tool -change "$hbDependency" "@executable_path/../Resources/lib/$dynDepOrigFile" "$1"
+            install_name_tool -change "$hbDependency" "@executable_path/../Resources/lib/$dynDepOrigFile" "$1"  || true
 
             # Check for loader path
             oToolLoader=$(otool -L "$1" 2>/dev/null | grep '@loader_path' | grep $dynDepOrigFile | cut -d\( -f1 | sed 's/^[[:blank:]]*//;s/[[:blank:]]*$//' ) || true
             if [[ -n "$oToolLoader" ]]; then
                 echo "Resetting loader path for dependency <$hbDependency> of <$1>"
                 oToolLoaderNew=$(echo $oToolLoader | sed "s#@loader_path/##" | sed "s#../../../../opt/.*##")
-                install_name_tool -change "$oToolLoader" "@loader_path/${oToolLoaderNew}${dynDepOrigFile}" "$1"
+                install_name_tool -change "$oToolLoader" "@loader_path/${oToolLoaderNew}${dynDepOrigFile}" "$1"  || true
             fi
         done
 
@@ -135,7 +135,7 @@ function reset_exec_path {
         echo "Resetting library ID of <$1>"
 
         # Set correct library id
-        install_name_tool -id "@executable_path/../Resources/lib/$libraryOrigFile" "$1"
+        install_name_tool -id "@executable_path/../Resources/lib/$libraryOrigFile" "$1"  || true
     fi
 }
 

